### ISSUE TITLE
The address of the bot comment does not contain the updated code.

### ISSUE THREAD

**Issue by warmthsea** [stackblitz-labs/pkg.pr.new#235]
PR create bot commit:
https://github.com/warmthsea/element-plus/pull/18#issuecomment-2336576214
(Does not include updated code)

PR update code:
https://github.com/warmthsea/element-plus/pull/18/checks?check_run_id=29967470452

<hr />

Does the original comment address need to be updated?

**Comment by Aslemammad** [comment]
Yep, the address of the pull request comment should not change since it addresses the latest commit of the PR always when you put the number of the PR there.

But the check run is what gets updated, when you visit the check run (`commit: 3b1av3f`) it contains the built package for that specific version.

**Comment by warmthsea** [comment]
> Yep, the address of the pull request comment should not change since it addresses the latest commit of the PR always when you put the number of the PR there.
> 
> But the check run is what gets updated, when you visit the check run (`commit: 3b1av3f`) it contains the built package for that specific hash.


But when executing ↓, the installed build product is the original one, not the latest one. Is this by design?

(18 represents pr floor)
```sh
pnpm add https://pkg.pr.new/warmthsea/element-plus@18
```

**Comment by Aslemammad** [comment]
I think this is because of caching! 

Can you try adding the -f flag and let me know!

**Comment by warmthsea** [comment]
> I think this is because of caching!
> 
> Can you try adding the -f flag and let me know!

Thank you very much for your reply. I have some doubts. Regarding whether the initial comment address of the installation is the latest code.

### Info:
 
**window 11**
npm origin: https://registry.npmjs.org/  (**default settings**)
use pnpm version: **9.9.0** 
node version: **v21.7.3**
chrome、firefox

test pr: https://github.com/warmthsea/element-plus/pull/20
test install repo: https://github.com/warmthsea/elp-pkg-pr-release-test
<hr />

### Step:
1、install pr Bot [first commit link](https://github.com/warmthsea/element-plus/pull/20#issuecomment-2342716389)
```sh
pnpm add https://pkg.pr.new/warmthsea/element-plus@20
```
2、update warmthsea/element-plus components 
3、**wait** for **pkg.pr.new** ci to complete
4、again install
```sh
pnpm add https://pkg.pr.new/warmthsea/element-plus@20
```

<hr />

1、use [Vercel](https://vercel.com/) deploy
 - execute in a short time ([demo](https://elp-pkg-pr-release-test-429xsnsl5-warmthseas-projects.vercel.app/)) (update content not effective)
 - execute in one minute ([demo](https://elp-pkg-pr-release-test.vercel.app/)) (effective)

2、run locally (dev)
-  all time (effective)

try: 
 - delete pnpm-lock.yaml 
 - again clone repo
 - delete node_modules
 - pnpm i xx --force

none of them took effect.

**Comment by Aslemammad** [comment]
Hmm, but as I curl, it addresses the latest commit! 

![image](https://github.com/user-attachments/assets/bddc1cc8-b993-45fd-a65f-47a3e1eb1b72)

**Comment by thewilloftheshadow** [comment]
I'm also experiencing this issue with https://pkg.pr.new/buape/carbon/@buape/carbon@126. It shows the correct commit hash when curling it and I tried using --force as well, but it is still not actually downloading the PR's code, instead it's using `main`'s code. 

Here is the GitHub Action that I'm running: https://github.com/buape/carbon/blob/main/.github/workflows/ci.yml

**Comment by thewilloftheshadow** [comment]
I was able to confirm with https://github.com/buape/carbon/pull/127 that it was using the code from `main`, as that changes PR does have main's code with the fixes for that version

**Comment by Aslemammad** [comment]
As far as I remember, `pull_request_target` uses the main branch 🤔

So maybe that's the issue!
